src/main/java/com/t13g2/forum/model/ForumBook.java

```diff
@@ -27,12 +28,12 @@
         persons = new UniquePersonList();
     }
 
-    public AddressBook() {}
+    public ForumBook() {}
```

Comment:
Actually after looking at the code, you guys completely skipped this layer of abstraction right? Data from commands is written directly to storage via the `UnitOfWork` interface.
`ModelManager` etc. are not being used anymore in your project right now AFAICS, but that shouldn't be the case. Have a look at how `Model` component wraps around the storage.

Reply:
Hi, for this, May I discuss with you about this? I've taken a brief look at `ModelManager`, this class defines what we can do with the data.
Eg:

* We can add a `Person` to storage
* We can update a `Person` to storage
* We can check if a `Person` exists 
* We CANNOT directly check if a `Person` with a specific attribute exists

This is what I mean by defining the behaviour.

This is a good design in AddressBook case since there is only one entity to manage, `Person`.

However, in our case, we have more entities, `Thread`, `Comment`, `User`. To better manage the data, `UnitOfWork` is implemented to manage them, it does not directly manage the data, instead, there is another layer of abstraction `EntityRepository`, these repositories interface defines how we can interact with data.

Eg:

* We can add `Announcement`
* We can add `User`
* We can get latest `Announcement`
* We CANNOT directly get `User` by its password

This is preciously what model manager does. Yes, we are not currently using `ModelManager`, because I don't see it fit. If I modify `ModelManager` to wrap `UnitOfWork`, it will end up more than 20 methods managing `Thread`, `Comment`, `User` etc. entities. This is obviously not a good design (single responsibility?).

TLDR:
`EntityRepository` is somehow similar to `ModelManager`, `UnitOfWork` is an additional layer of abstraction which takes care of data consistency(commit, rollback) and manages `EntityRepository`.

I'm not sure if I got this correctly, do let me know what I can improve on this. Thanks a lot for your effort.

Reply:
Hi good morning! @ewaldhew